app/controllers/my_orders.js

```diff
@@ -17,6 +18,37 @@ export default applicationController.extend({
   cancellationReasonWarning: false,
   applicationController: Ember.inject.controller('application'),
   hideHeaderBar: Ember.computed.alias("applicationController.hideHeaderBar"),
+  messageSortOptions: ['createdAt:asc'],
+  noMessage: Ember.computed.empty("selectedOrder.messages"),
+
+  displayChatNote: Ember.computed('noMessage', 'disabled', function () {
+    return this.get("noMessage") && !this.get("disabled");
+  }),
+
+  sortedMessages: Ember.computed.sort("selectedOrder.messages", "messageSortOptions"),
+
+  groupedMessages: Ember.computed("sortedMessages", function () {
+    return this.groupBy(this.get("sortedMessages"), "createdDate");
+  }),
+
+  groupBy: function (content, key) {
```

Comment:
note: we have lodash installed as a dependency, feel free to use it (e.g `_.groupBy(...)`)   😄

Reply:
Good suggestion

Reply:
sure @patrixr , i'll try to implement that way.